utils/lib/nr-graphql-helpers.ts

```diff
@@ -175,6 +175,32 @@ export const getCategoryTermsFromKeywords = async (
   return categoryKeywords.length > 0 ? categoryKeywords : undefined;
 };
 
+type CoreDataSourceSearchResults = {
+  actor: {
+    nr1Catalog: {
+      search: {
+        results: {
+          id: string;
+        }[];
+      };
+    };
+  };
+};
+
+export const getCoreDataSourceIds = async (): Promise<string[]> => {
```

Comment:
I think this name is a little misleading, since the query will actually return all data sources. Perhaps something like 
```suggestion
export const getPublishedDataSourceIds = async (): Promise<string[]> => {
```